packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -884,15 +891,24 @@ export interface SharedObjectKind<out TSharedObject = unknown>
 export function createSharedObjectKind<TSharedObject>(
 	factory: (new () => IChannelFactory<TSharedObject>) & { readonly Type: string },
 ): ISharedObjectKind<TSharedObject> & SharedObjectKind<TSharedObject> {
-	const result: ISharedObjectKind<TSharedObject> = {
+	const result: ISharedObjectKind<TSharedObject> &
+		Omit<SharedObjectKind<TSharedObject>, "brand"> = {
 		getFactory(): IChannelFactory<TSharedObject> {
 			return new factory();
 		},
 
 		create(runtime: IFluidDataStoreRuntime, id?: string): TSharedObject {
 			return runtime.createChannel(id, factory.Type) as TSharedObject;
 		},
+
+		is(value: IFluidLoadable): value is IFluidLoadable & TSharedObject {
+			return isChannel(value) && value.attributes.type === factory.Type;
+		},
 	};
 
 	return result as typeof result & SharedObjectKind<TSharedObject>;
 }
+
+function isChannel(loadable: IFluidLoadable): loadable is IChannel {
+	return (loadable as IChannel).attributes !== undefined;
```

Comment:
This isn't truly robust, as its quite possible for some other type to have an attributes field.
```suggestion
	// This assumes no other IFluidLoadable has an `attributes` field, and thus may not be fully robust.
	return (loadable as IChannel).attributes !== undefined;
```